src/lib/check-medic-conf-dependency-version.js

```diff
@@ -1,70 +1,59 @@
 /**
  * Attempt to check that the version of medic-conf that a project relies on
  * matches the version of medic-conf being used to configure that project.
- *
- * Currently this check is only performed as part of app-settings compilation;
- * if required in more places then we might move the check to the main CLI
- * script.
  */
 
-const error = require('./log').error;
 const fs = require('./sync-fs');
-const readline = require('readline-sync');
 const semver = require('semver');
 const { warn } = require('./log');
 
-module.exports = projectDir => {
-  const myVersion = require('../../package.json').version;
+const runningVersion = require('../../package.json').version;
 
-  const theirPackageVersion = readRequestedVersion(projectDir);
-  if(!theirPackageVersion) {
+module.exports = projectDir => {
+  const projectVersion = readRequestedVersion(projectDir);
+  if(!projectVersion) {
     warn('Project has no dependency on medic-conf.');
     return;
   }
 
-  if(!semver.satisfies(myVersion, theirPackageVersion)) {
-    throw new Error(`medic-conf version ${myVersion} does not match the project's required version: ${theirPackageVersion}`);
+  const majorRunningVersion = semver.major(runningVersion);
+  let upgradeDowngradeLocalMsg = '';
+  let upgradeDowngradeProjectMsg = '';
+  if(semver.satisfies(projectVersion, `<${majorRunningVersion}.x || >${runningVersion}`)) {
```

Comment:
consider pulling these two semver comparison expressions out into variables (either the text, or call `satisfies` and store the answer) so they can be reused in the lower `if` statements, to avoid repetition.